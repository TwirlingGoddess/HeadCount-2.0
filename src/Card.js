import React from 'react';
import PropTypes from 'prop-types';


const Card = ({ title, content, updateClickedCard }) => {
  const cardData =

    Object.keys(content).map((year, index) => {
      const toggle = content[year] <= .5 ? 'low' : 'high';
      return <aside 
        key={index} 
        className={toggle}> {year}: {content[year]} </aside>;

    });

  return (
    <div  className="Card" 
      onClick={() => updateClickedCard(title, content)}>
      <h3>{title}</h3>
      <ul>{cardData}</ul>
    </div>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  content: PropTypes.object,
  updateClickedCard: PropTypes.func
};

export default Card;