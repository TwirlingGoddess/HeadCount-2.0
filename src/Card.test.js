import React from 'react';
import Card from './Card.js';
import { shallow } from 'enzyme';

describe('Card', () => {
  it('should have some text inside a paragraph tags', () => {
    // setup
    const wrapper = shallow(<Card stuff="School"/>)
    const expected = <p>School</p>
    // execution

    // expectation
    expect(wrapper.contains(expected)).toEqual(true)
  })

})