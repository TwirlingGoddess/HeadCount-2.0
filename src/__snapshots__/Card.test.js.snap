// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    content={
        Object {
            "2004": 0.302,
            "2005": 0.267,
          }
    }
    title="ACADEMY 20"
    updateClickedCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          ACADEMY 20
</h3>,
        <ul>
          <aside
                    className="low"
          >
                     
                    2004
                    : 
                    0.302
                     
          </aside>
          <aside
                    className="low"
          >
                     
                    2005
                    : 
                    0.267
                     
          </aside>
</ul>,
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ACADEMY 20",
        },
        "ref": null,
        "rendered": "ACADEMY 20",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <aside
              className="low"
>
               
              2004
              : 
              0.302
               
</aside>,
            <aside
              className="low"
>
               
              2005
              : 
              0.267
               
</aside>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2004",
                ": ",
                0.302,
                " ",
              ],
              "className": "low",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2004",
              ": ",
              0.302,
              " ",
            ],
            "type": "aside",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "2005",
                ": ",
                0.267,
                " ",
              ],
              "className": "low",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "2005",
              ": ",
              0.267,
              " ",
            ],
            "type": "aside",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            ACADEMY 20
</h3>,
          <ul>
            <aside
                        className="low"
            >
                         
                        2004
                        : 
                        0.302
                         
            </aside>
            <aside
                        className="low"
            >
                         
                        2005
                        : 
                        0.267
                         
            </aside>
</ul>,
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ACADEMY 20",
          },
          "ref": null,
          "rendered": "ACADEMY 20",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <aside
                className="low"
>
                 
                2004
                : 
                0.302
                 
</aside>,
              <aside
                className="low"
>
                 
                2005
                : 
                0.267
                 
</aside>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2004",
                  ": ",
                  0.302,
                  " ",
                ],
                "className": "low",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2004",
                ": ",
                0.302,
                " ",
              ],
              "type": "aside",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "2005",
                  ": ",
                  0.267,
                  " ",
                ],
                "className": "low",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "2005",
                ": ",
                0.267,
                " ",
              ],
              "type": "aside",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
